File: app/src/main/java/com/example/map/base/BaseViewModel.java
package com.example.map.base;

import androidx.lifecycle.ViewModel;

import io.reactivex.rxjava3.disposables.CompositeDisposable;

public class BaseViewModel extends ViewModel {
    protected final CompositeDisposable compositeDisposable = new CompositeDisposable();

    public void onDestroy() {
        compositeDisposable.clear();
    }
}

File: app/src/main/java/com/example/map/base/BaseViewModel.kt
package com.example.map.base

import androidx.lifecycle.ViewModel
import io.reactivex.rxjava3.disposables.CompositeDisposable

open class BaseViewModel : ViewModel() {
    protected val compositeDisposable = CompositeDisposable()
    fun onDestroy() {
        compositeDisposable.clear()
    }
}
